torch/nn/attention/_fa4.py

```diff
@@ -0,0 +1,599 @@
+# mypy: allow-untyped-defs
+
+from __future__ import annotations
+
+import importlib
+from dataclasses import dataclass
+from typing import Any, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from types import ModuleType
+
+import torch
+from torch.library import Library
+
+
+__all__ = [
+    "register_flash_attention_fa4",
+    "flash_attention_fa4_status",
+]
+
+
+@dataclass
+class _FA4FlashAttentionState:
+    module_path: str = "flash_attn.cute.interface"
+    module: ModuleType | None = None
+    registered: bool = False
+    last_error: str | None = None
+
+
+_FA4_STATE = _FA4FlashAttentionState()
+_FA4_LIBRARY: Library | None = None
+
+
+def _flash_attention_forward_fallback(*args, **kwargs):
+    return torch.ops.aten._flash_attention_forward.default(*args, **kwargs)
```

Comment:
Infinite recursion inc :D 
You need to get the kernel below using Mikayla's API.

Reply:
none of this is ready for review, stop looking lol